File: src/main/java/com/statuswatchr/statuswatchr/model/Watchr.java
package com.statuswatchr.statuswatchr.model;

import jakarta.persistence.*;
import lombok.*;
import java.time.Instant;

@Getter
@Setter
@NoArgsConstructor
@AllArgsConstructor
@Builder
@Entity
@Table(name = "watchrs")
public class Watchr {
        @Id
        private Long id;
        private String name;
        private String url;
        private Integer intervalSeconds;
        private Status status;
        private Instant lastCheckedAt;
        private String lastError;

        @PrePersist
        public void prePersist() {
                if (status == null) status = Status.UNKNOWN;
                if (intervalSeconds == null) intervalSeconds = 60;
        }
}

File: statuswatchr-frontend/src/types/watchr.ts
export type WatchrStatus = "UP" | "DOWN" | "UNKNOWN";

export type Watchr = {
  id: number;
  name: string;
  url: string;
  intervalSeconds: number;
  status: WatchrStatus;
  lastCheckedAt: string;   // Instant becomes ISO string in JSON
  lastError: string | null; // could be null sometimes
};